src/mlpack/core/data/csv_parser_impl.hpp

```diff
@@ -141,60 +192,57 @@ namespace data
     return load_okay;		
   }
 
-  template<typename eT>
-  inline bool LoadCSV(const std::string& name, std::string& err_msg)
+  template <typename MatType>
+  bool LoadData(const std::string& name, MatType& x, const mlpack::data::file_type type)
   {
+    // bool load_okay = false;
+    std::string err_msg;
+    std::string print_status;
     std::fstream f;
-    arma::Mat<eT> x;
 
     f.open(name.c_str(), std::fstream::in);
 
-    bool load_okay = f.is_open();
-
-    if (load_okay == false)
+    switch (type)
     {
-      return false;
-    }
+      case mlpack::data::file_type::csv_ascii:
+        return LoadCSVV<MatType>(x, f, print_status);
 
-    if (load_okay)
-    {
-      load_okay = LoadCSV<eT>(x, f, err_msg);
-    }
+      case mlpack::data::file_type::file_type_unknown:
+        return true;
 
-    f.close();
+      case mlpack::data::file_type::auto_detect:
+        return true;
 
-    return load_okay;
-  }
+      case mlpack::data::file_type::raw_ascii:
```

Comment:
@shrit How should I handle the other types like `raw_ascii`?

Reply:
mlpack does handle `raw_ascii` when it is used for loading, this type is defined inside `detect_file_type.cpp` inside mlpack.
I do not think this switch case should be defined here. This file should be related only to parsing CSV files. :+1: 
I think you need to refactor the `detect_file_type.cpp` and have a translation shim there, that translate from `mlpack::data::file_type` to `armadillo::file_type`